clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jar/JarArtifactCredentials.java

```diff
@@ -0,0 +1,118 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Data
+public class JarArtifactCredentials implements ArtifactCredentials {
+  private final String name;
+  private final List<String> types = Collections.singletonList("jar/archive");
+
+  private final RepositorySystem repositorySystem;
+  private final RepositorySystemSession repositorySystemSession;
+  private final RemoteRepository remoteRepository;
+
+  public JarArtifactCredentials(JarArtifactAccount account) {
+    this.name = account.getName();
+    this.repositorySystem = getRepositorySystem();
+    this.repositorySystemSession = getRepositorySystemSession(this.repositorySystem);
+    this.remoteRepository = new RemoteRepository.Builder(account.getName(), "default", account.getRepositoryUrl()).build();
+  }
+
+  public InputStream download(Artifact artifact) {
+    File jarFile = getJarFile(artifact.getReference());
+    try {
+      return new FileInputStream(jarFile);
+    } catch (FileNotFoundException e) {
+      throw new IllegalStateException("File not found", e);
+    }
+  }
+
+  private RepositorySystem getRepositorySystem() {
+    DefaultServiceLocator locator = MavenRepositorySystemUtils.newServiceLocator();
+    locator.addService(RepositoryConnectorFactory.class, BasicRepositoryConnectorFactory.class);
+    locator.addService(TransporterFactory.class, FileTransporterFactory.class);
+    locator.addService(TransporterFactory.class, HttpTransporterFactory.class);
+    locator.addService(RepositoryLayoutProvider.class, DefaultRepositoryLayoutProvider.class);
+    return locator.getService(RepositorySystem.class);
+  }
+
+  private Path getRepositoryPath() throws IOException {
+    String uuid = UUID.randomUUID().toString();
+    Path tempDir = Paths.get(System.getProperty("java.io.tmpdir"), uuid);
+    if (!Files.isDirectory(tempDir)) {
+      Files.createDirectories(tempDir);
+    }
+    return tempDir;
+  }
+
+  private RepositorySystemSession getRepositorySystemSession(RepositorySystem system) {
+    DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+    Path tempDir;
+    try {
+      tempDir = getRepositoryPath();
+    } catch (IOException e) {
+      throw new IllegalStateException("Could not create temporary directory for jar artifact credentials", e);
+    }
+    LocalRepository localRepo = new LocalRepository(tempDir.toAbsolutePath().toString());
+    session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, localRepo));
+    return session;
+  }
+
+  private File getJarFile(String reference) {
+    org.eclipse.aether.artifact.Artifact jarArtifact = new DefaultArtifact(reference);
+    ArtifactRequest request = new ArtifactRequest(jarArtifact, Collections.singletonList(remoteRepository), null);
+    ArtifactResult result;
+    try {
+      result = repositorySystem.resolveArtifact(repositorySystemSession, request);
```

Comment:
Does this result in transitives downloading? I'll make sure we don't do that inadvertently in the Ivy impl.